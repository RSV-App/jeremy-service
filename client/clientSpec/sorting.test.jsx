import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import Sorting from '../src/components/Sorting.jsx';

Enzyme.configure({ adapter: new Adapter() });

describe('Sorting', () => {
  const tags = ['a', 'b'];
  const choosingSort = false;
  const options = ['x', 'y'];
  const handleSortClick = () => {};
  const handleSortOptionClick = () => {};

  it('renders the component', () => {
    const tree = renderer
      .create(
        <Sorting
          tags={tags}
          choosingSort={choosingSort}
          options={options}
          handleSortClick={handleSortClick}
          handleSortOptionClick={handleSortOptionClick}
        />
      ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the header text', () => {
    const wrapper = shallow(
      <Sorting
        tags={tags}
        choosingSort={choosingSort}
        options={options}
        handleSortClick={handleSortClick}
        handleSortOptionClick={handleSortOptionClick}
      />
    );
    const list = wrapper.find('div h4').first();
    expect(list.text()).toBe('Sort by');
  });
});
