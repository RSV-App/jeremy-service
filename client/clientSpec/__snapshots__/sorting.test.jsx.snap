// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sorting renders the component 1`] = `
<div
  className="sorting-panel"
>
  <h4>
    Sort by
  </h4>
  <div
    className="sort-area"
  >
    <div
      className="sort-dropdown"
      onClick={[Function]}
    >
      <span
        className="sorter-name"
      />
      <i
        className="sorter-icon"
      />
    </div>
    <div
      className="options-container"
    />
  </div>
  <h4>
    Filters
  </h4>
  <div
    className="filters"
  >
    <div
      className="filter-button"
      data-tag="a"
    >
      <input
        className="filter-checkbox"
        type="checkbox"
      />
      <span
        className="filter-name"
      >
        a
      </span>
    </div>
    <div
      className="filter-button"
      data-tag="b"
    >
      <input
        className="filter-checkbox"
        type="checkbox"
      />
      <span
        className="filter-name"
      >
        b
      </span>
    </div>
  </div>
</div>
`;
